.lintstagedrc.json

```diff
@@ -0,0 +1,4 @@
+{
+  "*.{css,less,scss,graphql,gql,html,js,jsx,json,md,markdown,mdown,mkdn,mdx,ts,tsx,vue,yaml,yml}": "prettier --write",
+  "*.js": "eslint --fix"
```

Comment:
That will run Prettier too because we use `eslint-plugin-prettier`. (How about removing it BTW?)

Reply:
(I think it convenient that a `yarn lint:eslint --fix` run will do the static check and formatting at the same time. Why remove it?)

Reply:
It's seems to be a bit slow, which is especially noticeable in VS Code. Also it's strange that the Prettier project uses ESLint to run Prettier. Eat your own dog food and all that.

Reply:
Okay, I understand. I don't have strong opinion. If there is no other objection, let's remove it.

Reply:
See also https://github.com/prettier/prettier/issues/7828

Reply:
Yes, it's slow in vscode, but  easier to setup.

If we remove `eslint-plugin-prettier` we'll need run/config  both `eslint` and `prettier` to auto format code  in  `vscode`.

How about keep it, but exclude running prettier on `.js` from `lint-staged`.

Reply:
> we'll need run/config both eslint and prettier to auto format code in vscode.

Is it a problem? It works fine for me in other projects. Actually, it works better (faster) then ESLint with eslint-plugin-prettier.

Reply:
Thanks for the tip, I can try to run this way.

I guess we can remove `eslint-plugin-prettier`.

Reply:
@thorn0 

I tried you suggestion,

```
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true,
  },
  "editor.codeActionsOnSave": [
    "source.fixAll.eslint",
  ],
```

and tried to edit out `print-estree.js`, don't know if it became faster or now, both slow :).

But found difference, if I want undo, before, I only need undo once, now I need undo twice. Is my settings correct?

Reply:
Played with it too. You're right. Both setups are slow. Turns out in the Prettier project, I actually had both extensions enabled on save, which led to extra slowness. After disabling `editor.formatOnSave` it became more bearable. Seems like `eslint-plugin-prettier` stays.

Reply:
Do you have the same `undo` problem as me?

Reply:
Yes, I have. Somehow I didn't notice it before.

Reply:
Regarding, eating your own dogfood, we are doing it, since we are using the plugin! Are you guys want to remove ESLint completely or what?

Reply:
We discussed removing only `eslint-plugin-prettier`, not ESLint completely. The current setup doesn't use `prettier --check` and `prettier --write`. That's what I meant WRT dogfooding.